packages/node-utils/src/http.ts

```diff
@@ -364,8 +364,13 @@ const checkOrigin = ({
 
     if (origin) {
         isOriginAllowed = origins.some(o => {
-            // match from the end to allow subdomains
-            return new URL(origin).hostname.endsWith(new URL(o).hostname);
+            try {
```

Comment:
I am adding this try catch to avoid the node server to crash when origin is defined but it is not a parseable URL like in the case of default Tor when origin is "null" as a string.